src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -268,16 +268,42 @@ public String getTypeDescription() {
       help = "Whether to upload locally executed action results to the remote cache.")
   public boolean remoteUploadLocalResults;
 
+  @Deprecated
   @Option(
       name = "incompatible_remote_build_event_upload_respect_no_cache",
       defaultValue = "false",
       documentationCategory = OptionDocumentationCategory.REMOTE,
       effectTags = {OptionEffectTag.UNKNOWN},
+      deprecationWarning =
+          "--incompatible_remote_build_event_upload_respect_no_cache has been deprecated in favor"
+              + " of --experimental_remote_build_event_upload=minimal.",
       help =
           "If set to true, outputs referenced by BEP are not uploaded to remote cache if the"
               + " generating action cannot be cached remotely.")
   public boolean incompatibleRemoteBuildEventUploadRespectNoCache;
 
+  @Option(
+      name = "experimental_remote_build_event_upload",
+      defaultValue = "all",
+      documentationCategory = OptionDocumentationCategory.REMOTE,
+      effectTags = {OptionEffectTag.UNKNOWN},
+      converter = RemoteBuildEventUploadModeConverter.class,
+      help =
+          "If set to 'minimal', local outputs are not uploaded to the remote cache, except for"
+              + " files that are important to the consumers of BEP (e.g. test logs and timing"
+              + " profile). file:// scheme is used for the paths of local files and bytestream://"
+              + " scheme is used for paths of (already) uploaded files. Default to"
```

Comment:
Do we need to mention the path behavior of `all`? Or adjust it to be agnostic of the setting (since the previous sentence already states what is or is not being uploaded)?

Reply:
Done.